### PR TITLE
chore: refactor dialogs

### PR DESCRIPTION
**Pull request by presleyp**
Just a refactor to lay the groundwork for #3384.

- put all dialogs in one directory
- repurpose the one-off component for deleting a workspace into a general component for deleting things, used for templates and workspaces (later this can become a component that asks you to type in the name of what you want to delete, for extra safety)

I've checked that this leaves the delete terminal dialog in a good state but I'm blocked on checking the delete workspace dialog right now.

### CODE REVIEW

File: site/src/components/Dialogs/DeleteDialog/DeleteDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { ReactNode } from "react"
+import { ConfirmDialog } from "../ConfirmDialog/ConfirmDialog"
+
+export interface DeleteDialogProps {
+  isOpen: boolean
+  handleConfirm: () => void
```
Comment (BrunoQuaresma) [inline site/src/components/Dialogs/DeleteDialog/DeleteDialog.tsx:6]: Why not keep using the `onSomeAction` pattern? I think this is the most used one from libs.

**Review by BrunoQuaresma — APPROVED** [review]
I'm ok with the changes. Just two minor stuff:
- A question about the "pattern" that we want to use for event handlers
- Idk but the diffs on Storybook don't look correct - not showing buttons. Idk if it is something fixable so I will let the approval to you

File: site/src/components/Dialogs/DeleteDialog/DeleteDialog.tsx (hunk 1 above)
Comment (presleyp) [reply site/src/components/Dialogs/DeleteDialog/DeleteDialog.tsx:6]: The `DeleteWorkspaceDialog` component used `handle`, but I'd like it to be consistent so I'll switch it

**Comment by presleyp** [comment]
@BrunoQuaresma yeah I've seen this storybook issue enough times now that I'm not concerned that it's a problem in the actual app, but I'll have to try to figure out what the problem is.

**Pull request merged**